src/components/ui4/CarouselUi4.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import { View } from 'react-native'
+import Carousel, { Pagination } from 'react-native-snap-carousel'
+
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+
+interface Props {
+  children: React.ReactNode | React.ReactNode[]
+  height: number
+  width: number
+}
+
+const DOT_SIZE_REM = 0.5
+
+/**
+ * A horizontal carousel with pagination dots
+ */
+export const CarouselUi4 = (props: Props) => {
+  const { children, height, width } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
+  const nonNullChildren = React.Children.toArray(children).filter(child => child != null && React.isValidElement(child))
+  const numChildren = nonNullChildren.length
+
+  const [activeIndex, setActiveIndex] = useState(0)
+
+  const renderItem = ({ item }: { item: React.ReactNode }) => <View style={[styles.childContainer, { width: width * 0.9, height }]}>{item}</View>
+
+  if (children == null || numChildren === 0) return null
+
+  return (
+    <View>
```

Comment:
Does the `View` do anything? Could you use `<>` instead?

If it's a drop-in replacement, great! If it's not, don't waste any time on this one.